Source/WebKit/webpushd/WebClipCache.h

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (C) 2024 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#if ENABLE(WEB_PUSH_NOTIFICATIONS) && PLATFORM(IOS)
+
+#include <WebCore/SecurityOriginData.h>
+#include <tuple>
+#include <wtf/HashMap.h>
+#include <wtf/HashSet.h>
+#include <wtf/TZoneMalloc.h>
+
+namespace WebPushD {
+
+class WebClipCache {
+    WTF_MAKE_TZONE_ALLOCATED(WebClipCache);
+public:
+    WebClipCache(const String& path);
+
+    String preferredWebClipIdentifier(const String& bundleIdentifier, const WebCore::SecurityOriginData&);
+    HashSet<String> visibleWebClipIdentifiers(const String& bundleIdentifier);
+    bool isWebClipVisible(const String& bundleIdentifier, const String& webClipIdentifier);
```

Comment:
Can these methods be const?

Reply:
In theory yes, although e.g. `preferredWebClipIdentifier` actually mutates some stuff behind the scenes. But I can use `mutable` to fix that.